zaza/openstack/charm_tests/cloudkitty/tests.py

```diff
@@ -44,53 +43,86 @@ def setUpClass(cls):
-        """Test enabling hashmap module via API."""
-        logging.info('Enabling hashmap module')
+    def test_401_module_enable_and_disable(self):
+        """Test enable and disable module via API."""
+        enable = True
+        disable = False
+
         rating = self.cloudkitty.rating
-        rating.update_module(module_id='hashmap', enabled=True)
 
-        hashmap = rating.get_module(module_id='hashmap')
-        assert hashmap.get('enabled')
+        logging.info('Enabling hashmap module.')
+        rating.update_module(module_id='hashmap', enabled=enable)
+
+        module = rating.get_module(module_id='hashmap')
+        assert module.get('enabled') == enable
+
+        logging.info('Disabling hashmap module.')
+        rating.update_module(module_id='hashmap', enabled=disable)
+
+        module = rating.get_module(module_id='hashmap')
+        assert module.get('enabled') == disable
 
     def test_402_service_create_and_delete(self):
         """Test service create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
         hashmap = self.cloudkitty.rating.hashmap
         service = hashmap.create_service(name='test')
 
-        assert service['name'] == 'test'
+        assert service.get('name') == 'test'
 
-        service_id = service['service_id']
+        service_id = service.get('service_id')
         hashmap.delete_service(service_id=service_id)
 
     def test_403_field_create_and_delete(self):
         """Test field create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
         hashmap = self.cloudkitty.rating.hashmap
         service = hashmap.create_service(name='test')
 
-        service_id = service['service_id']
+        service_id = service.get('service_id')
         field = hashmap.create_field(name='test', service_id=service_id)
 
-        field_id = field['field_id']
+        field_id = field.get('field_id')
         hashmap.delete_field(field_id=field_id)
         hashmap.delete_service(service_id=service_id)
 
     def test_404_mapping_create_and_delete(self):
         """Test mapping create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
         hashmap = self.cloudkitty.rating.hashmap
         service = hashmap.create_service(name='test')
 
-        service_id = service['service_id']
+        service_id = service.get('service_id')
         field = hashmap.create_field(name='test', service_id=service_id)
 
-        field_id = field['field_id']
+        field_id = field.get('field_id')
         mapping = hashmap.create_mapping(
             type='flat', field_id=field_id, value='test', cost=0.1)
 
-        mapping_id = mapping['mapping_id']
+        mapping_id = mapping.get('mapping_id')
         hashmap.delete_mapping(mapping_id=mapping_id)
```

Comment:
Hmm, this is all cleanup stuff right? This is generally done within the test case itself to do the cleanup - which is handled even in the event of certain errors, etc. We should push the deletes/cleanups to the cleanup functions.

Reply:
created a teardown function to handle this ;)